{"content":"---\ntitle: \"Compiler Warning (level 1) CS1058\"\nms.date: 07/20/2015\nf1_keywords: \n  - \"CS1058\"\nhelpviewer_keywords: \n  - \"CS1058\"\nms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d\n---\n# Compiler Warning (level 1) CS1058\nA previous catch clause already catches all exceptions. All exceptions thrown will be wrapped in a System.Runtime.CompilerServices.RuntimeWrappedException  \n  \n This attribute causes CS1058 if a `catch()` block has no specified exception type after a `catch (System.Exception e)` block. The warning advises that the `catch()` block will not catch any exceptions.  \n  \n A `catch()` block after a `catch (System.Exception e)` block can catch non-CLS exceptions if the `RuntimeCompatibilityAttribute` is set to false in the AssemblyInfo.cs file: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`. If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the `catch (System.Exception e)` block catches them. For more information, see [How to: Catch a non-CLS Exception](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md).  \n  \n## Example  \n The following example generates CS1058.  \n  \n```csharp  \n// CS1058.cs  \n// CS1058 expected  \nusing System.Runtime.CompilerServices;  \n  \n// the following attribute is set to true by default in the C# compiler  \n// set to false in your source code to resolve CS1058  \n[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = true)]  \n  \nclass TestClass   \n{  \n   static void Main()   \n   {  \n      try {}  \n  \n      catch (System.Exception e) {   \n         System. Console.WriteLine(\"Caught exception {0}\", e);  \n      }  \n  \n      catch {}   // CS1058. This line will never be reached.  \n   }  \n}  \n```\n","nodes":[{"pos":[4,177],"embed":true,"restype":"x-metadata","content":"title: \"Compiler Warning (level 1) CS1058\"\nms.date: 07/20/2015\nf1_keywords: \n  - \"CS1058\"\nhelpviewer_keywords: \n  - \"CS1058\"\nms.assetid: ed50590c-f130-47c3-976d-322a6c8f996d","nodes":[{"content":"Compiler Warning (level 1) CS1058","nodes":[{"pos":[0,33],"content":"Compiler Warning (level 1) CS1058","nodes":[{"content":"Compiler Warning (level 1) CS1058","pos":[0,33]}]}],"path":["title"],"nosxs":false}],"yml":true},{"pos":[184,217],"content":"Compiler Warning (level 1) CS1058","linkify":"Compiler Warning (level 1) CS1058","nodes":[{"content":"Compiler Warning (level 1) CS1058","pos":[0,33]}]},{"content":"A previous catch clause already catches all exceptions.","pos":[218,273]},{"content":"All exceptions thrown will be wrapped in a System.Runtime.CompilerServices.RuntimeWrappedException","pos":[274,372]},{"content":"This attribute causes CS1058 if a <ph id=\"ph1\">`catch()`</ph> block has no specified exception type after a <ph id=\"ph2\">`catch (System.Exception e)`</ph> block.","pos":[379,504],"source":"This attribute causes CS1058 if a `catch()` block has no specified exception type after a `catch (System.Exception e)` block."},{"content":"The warning advises that the <ph id=\"ph1\">`catch()`</ph> block will not catch any exceptions.","pos":[505,580],"source":" The warning advises that the `catch()` block will not catch any exceptions."},{"content":"A <ph id=\"ph1\">`catch()`</ph> block after a <ph id=\"ph2\">`catch (System.Exception e)`</ph> block can catch non-CLS exceptions if the <ph id=\"ph3\">`RuntimeCompatibilityAttribute`</ph> is set to false in the AssemblyInfo.cs file: <ph id=\"ph4\">`[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`</ph>.","pos":[587,837],"source":"A `catch()` block after a `catch (System.Exception e)` block can catch non-CLS exceptions if the `RuntimeCompatibilityAttribute` is set to false in the AssemblyInfo.cs file: `[assembly: RuntimeCompatibilityAttribute(WrapNonExceptionThrows = false)]`."},{"content":"If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the <ph id=\"ph1\">`catch (System.Exception e)`</ph> block catches them.","pos":[838,1000],"source":" If this attribute is not set explicitly to false, all thrown non-CLS exceptions are wrapped as Exceptions and the `catch (System.Exception e)` block catches them."},{"content":"For more information, see <bpt id=\"p1\">[</bpt>How to: Catch a non-CLS Exception<ept id=\"p1\">](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md)</ept>.","pos":[1001,1145],"source":" For more information, see [How to: Catch a non-CLS Exception](../../../csharp/programming-guide/exceptions/how-to-catch-a-non-cls-exception.md)."},{"pos":[1154,1161],"content":"Example","linkify":"Example","nodes":[{"content":"Example","pos":[0,7]}]},{"content":"The following example generates CS1058.","pos":[1165,1204]}]}